### DISCUSSION TITLE
teslamate failed to return at extracting stats after the car was actually in offline mode for 24 hrs

### DISCUSSION THREAD

**Discussion by manikoutai** [teslamate-org/teslamate#2614] [Q&A]
Hi,

I am using teslamate for a few weeks only and this week we used our new tesla for the first "road trip" between Montreal and Toronto, which is about 500km each way.  As teslamate is running on a raspberry pi inside my house, and not exposed to be accessible outside of the house, I was only able to see my teslamate stats once I returned home.

Teslamate was only able to get the stats from Montreal to Toronto only and failed to get stats from returning home.

The car was actually parked on the second level of underground parking for about 24hrs. During that time it was normal for Teslamate to consider the car to be offline.  Meanwhile, teslamate failed to start to get new data once the car was back on the road. About 9hrs after being returned home, I realized the fact so I restart testlamate and "woke" the car, and only at that moment teslamate was able to grab new information from the car.

I can`t be 100% sure the reason was due to the actual offline state of the car while being parked as I am not familiar with the logs but I would be happy to share any files/logs if someone would like to help better understand the situation before we could consider this situation as a bug.

I also use the cloud application eevee.  This application was able to get from tesla all the information about the car before and after the official OFFLINE state.

Let me know if someone would be interested to have access to some log files to investigate.

Update 25/04: I have attached the log file as teslamate_log.txt I have also attached the screenshot of the offline situation
[teslamate_log.txt](https://github.com/adriankumpf/teslamate/files/8557198/teslamate_log.txt)
![state_offline_1](https://user-images.githubusercontent.com/103542406/165154746-1da77a4a-dda5-4966-8ba9-5c22d72a9a3b.jpg)

**Comment by cwanja** [comment]
Post the logs.

**Reply by manikoutai** [reply]
Hi,

 I just found out how to extract the log file and I have uploaded the file inside of the main post.

Let me know if you need anything else.

**Reply by cwanja** [reply]
Short of a detailed explanation, it looks like your car went offline and then the tokens expired. The refresh that is supposed to refresh your token was failing. This was the last failed attempt, before five minutes later it worked.

Failed attempt:
```
[36mteslamate_1  |[0m 2022-04-23 10:25:44.772 [info] Refreshing access token ...
[36mteslamate_1  |[0m 2022-04-23 10:25:52.790 [error] POST https://auth.tesla.com/oauth2/v3/token -> error: "non-existing domain" (8013.825 ms)
[36mteslamate_1  |[0m 2022-04-23 10:25:52.792 [warning] Token refresh failed: %TeslaApi.Error{
[36mteslamate_1  |[0m   env: nil,
[36mteslamate_1  |[0m   message: "non-existing domain",
[36mteslamate_1  |[0m   reason: :token_refresh
[36mteslamate_1  |[0m }
[36mteslamate_1  |[0m 2022-04-23 10:25:52.793 [warning] Retrying in 5 minutes...
```

Next attempt and got a token:
```
[36mteslamate_1  |[0m 2022-04-23 11:42:50.992 car_id=1 [error] Error / :unauthorized
[36mteslamate_1  |[0m 2022-04-23 11:42:50.991 [info] Refreshing access token ...
[36mteslamate_1  |[0m 2022-04-23 11:42:51.596 [info] POST https://auth.tesla.com/oauth2/v3/token -> 200 (607.689 ms)
[36mteslamate_1  |[0m 2022-04-23 11:42:51.630 [info] Scheduling token refresh in 6 h
[36mteslamate_1  |[0m 2022-04-23 11:43:21.170 car_id=1 [info] Start / :asleep
```

Did you change your password or anything and then update the tokens? 
```
[36mteslamate_1  |[0m 2022-04-21 19:13:38.755 [error] ** (RuntimeError) Client Error: %{"error_type" => "client_error", "msg_type" => "data:error", "tag" => "276117568696", "value" => "Can't validate token. "}
[36mteslamate_1  |[0m     (teslamate 1.26.1) lib/tesla_api/stream.ex:165: TeslaApi.Stream.handle_frame/2
[36mteslamate_1  |[0m     (websockex 0.4.3) lib/websockex.ex:1113: WebSockex.try_callback/3
[36mteslamate_1  |[0m     (websockex 0.4.3) lib/websockex.ex:797: WebSockex.common_handle/4
[36mteslamate_1  |[0m     (stdlib 3.17) proc_lib.erl:226: :proc_lib.init_p_do_apply/3
[36mteslamate_1  |[0m 
[36mteslamate_1  |[0m 2022-04-21 19:13:39.005 car_id=1 [info] Start / :offline
```

Did you drive to somewhere with no cellphone connection? I also see this in the logs:
```
[36mteslamate_1  |[0m 2022-04-21 18:48:09.039 [error] Vehicle Error: Vehicle is offline
[36mteslamate_1  |[0m 2022-04-21 18:48:13.747 [info] GET https://owner-api.teslamotors.com/api/1/vehicles/1492656516736953/vehicle_data -> 408 (172.894 ms)
[36mteslamate_1  |[0m 2022-04-21 18:48:13.955 car_id=1 [warning] Vehicle went offline while driving
```

**Reply by manikoutai** [reply]
Hi, thanks for looking into this.

I haven't done anything particular, to be honest as I was away from home.  The reason why the car was offline for about 24hrs make sense to me as it was under 2 level of underground parking, but during those 24hrs I haven't done anything particular with my Tesla account or "token".  Is there a link between the teslamate token and what the Eevee apps use to access the car's data since Eevee never faced any apparent loss of data i.e I can see all my drives toward home in eevee.

While extracting the data today, I saw another situation where the car appeared offline this week-end, but no consequences as the car wasn't used during such period. I have just attached the screenshot.

I am really puzzled since it was the first time I was using the car for a long period of time without accessing teslamate as its instance is not visible outside of my home.

As I review your logs extraction you seem to be working on a different situation/occurrence since the original problem started on April 21st, and end on April 23rd at 7:43:21AM.

Could something from my home internet lead to this situation?  I doubt it since I would expect a different error to be reported in such case.

Strange situation
![tesla_offline_2](https://user-images.githubusercontent.com/103542406/165183142-d7abc0b7-9cdd-488a-bb01-67b9dfb2b47c.jpg)
.
![tesla_online_1](https://user-images.githubusercontent.com/103542406/165184035-03333e9c-51d3-4b2e-bb26-963da8e2e647.jpg)

**Reply by manikoutai** [reply]
Here is a longer view of the states observe between apr 21st to apr 25th.

![Apr-21_TO_Apr-25](https://user-images.githubusercontent.com/103542406/165184265-0084273b-a056-4f3d-b658-d0c08c521da0.jpg)
d

**Reply by cwanja** [reply]
> Is there a link between the teslamate token and what the Eevee apps use to access the car's data since Eevee never faced any apparent loss of data i.e I can see all my drives toward home in eevee.

No clue what Eevee uses. If you are using two loggers, it could be conflicting. 

> As I review your logs extraction you seem to be working on a different situation/occurrence since the original problem started on April 21st, and end on April 23rd at 7:43:21AM.

Unless you have your database / containers set to the right timezone, the log timeline will be different from what Grafana uses (browser time) 

> Could something from my home internet lead to this situation? I doubt it since I would expect a different error to be reported in such case.

Do you have anything else running on the device that could be intermittently blocking the internet access? Or DNS routing software? 

Beyond that, really not sure. The `408` error really just saying that TeslaMate cannot access the Tesla API. Not much else we can tell.

**Reply by manikoutai** [reply]
I have some doubt that using more than 1 service to query the Tesla API could cause problems, normally those API scale very well are isolated.

The question remains: why teslamate was not able to recover after my car was in OFFLINE mode for over 24hrs.

**Reply by cwanja** [reply]
Do you run anything else on the Raspberry PI that could have conflicted with TeslaMate? 

>The question remains: why teslamate was not able to recover after my car was in OFFLINE mode for over 24hrs.

The reason is because the device could not connect to the API endpoint.

Your drive started and then the vehicle went offline. 
```
[36mteslamate_1  |[0m 2022-04-21 18:42:55.104 car_id=1 [info] Driving / Start
[36mteslamate_1  |[0m 2022-04-21 18:45:55.337 [info] GET https://owner-api.teslamotors.com/api/1/vehicles/1492656516736953/vehicle_data -> 408 (9194.391 ms)
[36mteslamate_1  |[0m 2022-04-21 18:45:55.338 [warning] TeslaApi.Error / %{"error" => "{\"error\": \"timeout\"}", "error_description" => "", "response" => nil}
[36mteslamate_1  |[0m 2022-04-21 18:45:55.339 car_id=1 [error] Error / :unknown
[36mteslamate_1  |[0m 2022-04-21 18:45:58.988 [error] Vehicle Error: Vehicle is offline
[36mteslamate_1  |[0m 2022-04-21 18:46:46.766 [info] GET https://owner-api.teslamotors.com/api/1/vehicles/1492656516736953/vehicle_data -> 408 (9186.213 ms)
[36mteslamate_1  |[0m 2022-04-21 18:46:46.767 [warning] TeslaApi.Error / %{"error" => "{\"error\": \"timeout\"}", "error_description" => "", "response" => nil}
[36mteslamate_1  |[0m 2022-04-21 18:46:46.767 car_id=1 [error] Error / :unknown
[36mteslamate_1  |[0m 2022-04-21 18:47:05.965 [info] GET https://owner-api.teslamotors.com/api/1/vehicles/1492656516736953/vehicle_data -> 408 (9193.922 ms)
[36mteslamate_1  |[0m 2022-04-21 18:47:05.967 [warning] TeslaApi.Error / %{"error" => "{\"error\": \"timeout\"}", "error_description" => "", "response" => nil}
[36mteslamate_1  |[0m 2022-04-21 18:47:05.968 car_id=1 [error] Error / :unknown
[36mteslamate_1  |[0m 2022-04-21 18:47:06.909 car_id=1 [info] Received stream data: :inactive
[36mteslamate_1  |[0m 2022-04-21 18:47:25.162 [info] GET https://owner-api.teslamotors.com/api/1/vehicles/1492656516736953/vehicle_data -> 408 (9190.792 ms)
[36mteslamate_1  |[0m 2022-04-21 18:47:25.164 [warning] TeslaApi.Error / %{"error" => "{\"error\": \"timeout\"}", "error_description" => "", "response" => nil}
[36mteslamate_1  |[0m 2022-04-21 18:47:25.165 car_id=1 [error] Error / :unknown
[36mteslamate_1  |[0m 2022-04-21 18:47:30.011 [notice]     :alarm_handler: {:set, {TeslaMate.Vehicles.Vehicle_1_api_error, :fuse_blown}}
[36mteslamate_1  |[0m 2022-04-21 18:47:44.374 [info] GET https://owner-api.teslamotors.com/api/1/vehicles/1492656516736953/vehicle_data -> 408 (9206.122 ms)
[36mteslamate_1  |[0m 2022-04-21 18:47:44.376 [warning] TeslaApi.Error / %{"error" => "{\"error\": \"timeout\"}", "error_description" => "", "response" => nil}
[36mteslamate_1  |[0m 2022-04-21 18:47:44.377 car_id=1 [error] Error / :unknown
[36mteslamate_1  |[0m 2022-04-21 18:48:03.569 [info] GET https://owner-api.teslamotors.com/api/1/vehicles/1492656516736953/vehicle_data -> 408 (9189.122 ms)
[36mteslamate_1  |[0m 2022-04-21 18:48:03.571 [warning] TeslaApi.Error / %{"error" => "{\"error\": \"timeout\"}", "error_description" => "", "response" => nil}
[36mteslamate_1  |[0m 2022-04-21 18:48:03.571 car_id=1 [error] Error / :unknown
[36mteslamate_1  |[0m 2022-04-21 18:48:09.039 [error] Vehicle Error: Vehicle is offline
[36mteslamate_1  |[0m 2022-04-21 18:48:13.747 [info] GET https://owner-api.teslamotors.com/api/1/vehicles/1492656516736953/vehicle_data -> 408 (172.894 ms)
[36mteslamate_1  |[0m 2022-04-21 18:48:13.955 car_id=1 [warning] Vehicle went offline while driving
[36mteslamate_1  |[0m 2022-04-21 18:48:19.146 [info] GET https://owner-api.teslamotors.com/api/1/vehicles/1492656516736953/vehicle_data -> 408 (187.554 ms)
[36mteslamate_1  |[0m 2022-04-21 18:48:24.546 [info] GET https://owner-api.teslamotors.com/api/1/vehicles/1492656516736953/vehicle_data -> 408 (199.495 ms)
[36mteslamate_1  |[0m 2022-04-21 18:48:29.942 [info] GET https://owner-api.teslamotors.com/api/1/vehicles/1492656516736953/vehicle_data -> 408 (188.460 ms)
[36mteslamate_1  |[0m 2022-04-21 18:48:35.576 [info] GET https://owner-api.teslamotors.com/api/1/vehicles/1492656516736953/vehicle_data -> 408 (442.545 ms)
[36mteslamate_1  |[0m 2022-04-21 18:48:39.833 [error] Vehicle Error: Vehicle is offline
```

Then it was unable to refresh the token:
```
[36mteslamate_1  |[0m 2022-04-21 19:13:08.012 [error] Vehicle Error: Vehicle is offline
[36mteslamate_1  |[0m 2022-04-21 19:13:17.507 [info] GET https://owner-api.teslamotors.com/api/1/vehicles/1492656516736953/vehicle_data -> 408 (174.675 ms)
[36mteslamate_1  |[0m 2022-04-21 19:13:38.755 [error] ** (RuntimeError) Client Error: %{"error_type" => "client_error", "msg_type" => "data:error", "tag" => "276117568696", "value" => "Can't validate token. "}
[36mteslamate_1  |[0m     (teslamate 1.26.1) lib/tesla_api/stream.ex:165: TeslaApi.Stream.handle_frame/2
[36mteslamate_1  |[0m     (websockex 0.4.3) lib/websockex.ex:1113: WebSockex.try_callback/3
[36mteslamate_1  |[0m     (websockex 0.4.3) lib/websockex.ex:797: WebSockex.common_handle/4
[36mteslamate_1  |[0m     (stdlib 3.17) proc_lib.erl:226: :proc_lib.init_p_do_apply/3
[36mteslamate_1  |[0m 
[36mteslamate_1  |[0m 2022-04-21 19:13:39.005 car_id=1 [info] Start / :offline
```

Then TeslaMate could not access the API endpoint.
```
[36mteslamate_1  |[0m 2022-04-22 23:00:53.854 [warning] Retrying in 5 minutes...
[36mteslamate_1  |[0m 2022-04-22 23:00:54.546 [error] GET https://owner-api.teslamotors.com/api/1/vehicles/1492656516736953 -> error: "non-existing domain" (8013.844 ms)
[36mteslamate_1  |[0m 2022-04-22 23:00:54.547 [warning] TeslaApi.Error / non-existing domain
[36mteslamate_1  |[0m 2022-04-22 23:00:54.547 car_id=1 [error] Error / :unknown
[36mteslamate_1  |[0m 2022-04-22 23:01:32.567 [error] GET https://owner-api.teslamotors.com/api/1/vehicles/1492656516736953 -> error: "non-existing domain" (8014.068 ms)
[36mteslamate_1  |[0m 2022-04-22 23:01:32.567 [warning] TeslaApi.Error / non-existing domain
[36mteslamate_1  |[0m 2022-04-22 23:01:32.568 car_id=1 [error] Error / :unknown
[36mteslamate_1  |[0m 2022-04-22 23:02:10.585 [error] GET https://owner-api.teslamotors.com/api/1/vehicles/1492656516736953 -> error: "non-existing domain" (8013.787 ms)
[36mteslamate_1  |[0m 2022-04-22 23:02:10.586 [warning] TeslaApi.Error / non-existing domain
[36mteslamate_1  |[0m 2022-04-22 23:02:10.586 car_id=1 [error] Error / :unknown
[36mteslamate_1  |[0m 2022-04-22 23:02:48.605 [error] GET https://owner-api.teslamotors.com/api/1/vehicles/1492656516736953 -> error: "non-existing domain" (8013.880 ms)
[36mteslamate_1  |[0m 2022-04-22 23:02:48.606 [warning] TeslaApi.Error / non-existing domain
[36mteslamate_1  |[0m 2022-04-22 23:02:48.606 car_id=1 [error] Error / :unknown
[36mteslamate_1  |[0m 2022-04-22 23:03:26.624 [error] GET https://owner-api.teslamotors.com/api/1/vehicles/1492656516736953 -> error: "non-existing domain" (8014.046 ms)
[36mteslamate_1  |[0m 2022-04-22 23:03:26.625 [warning] TeslaApi.Error / non-existing domain
[36mteslamate_1  |[0m 2022-04-22 23:03:26.625 car_id=1 [error] Error / :unknown
[36mteslamate_1  |[0m 2022-04-22 23:04:04.643 [error] GET https://owner-api.teslamotors.com/api/1/vehicles/1492656516736953 -> error: "non-existing domain" (8013.897 ms)
[36mteslamate_1  |[0m 2022-04-22 23:04:04.644 [warning] TeslaApi.Error / non-existing domain
[36mteslamate_1  |[0m 2022-04-22 23:04:04.644 car_id=1 [error] Error / :unknown
[36mteslamate_1  |[0m 2022-04-22 23:04:42.662 [error] GET https://owner-api.teslamotors.com/api/1/vehicles/1492656516736953 -> error: "non-existing domain" (8013.950 ms)
[36mteslamate_1  |[0m 2022-04-22 23:04:42.663 [warning] TeslaApi.Error / non-existing domain
[36mteslamate_1  |[0m 2022-04-22 23:04:42.663 car_id=1 [error] Error / :unknown
[36mteslamate_1  |[0m 2022-04-22 23:05:20.685 [error] GET https://owner-api.teslamotors.com/api/1/vehicles/1492656516736953 -> error: "non-existing domain" (8017.884 ms)
[36mteslamate_1  |[0m 2022-04-22 23:05:20.686 [warning] TeslaApi.Error / non-existing domain
[36mteslamate_1  |[0m 2022-04-22 23:05:20.686 car_id=1 [error] Error / :unknown
[36mteslamate_1  |[0m 2022-04-22 23:05:53.856 [info] Refreshing access token ...
[36mteslamate_1  |[0m 2022-04-22 23:05:58.704 [error] GET https://owner-api.teslamotors.com/api/1/vehicles/1492656516736953 -> error: "non-existing domain" (8013.802 ms)
[36mteslamate_1  |[0m 2022-04-22 23:05:58.705 [warning] TeslaApi.Error / non-existing domain
[36mteslamate_1  |[0m 2022-04-22 23:05:58.705 car_id=1 [error] Error / :unknown
[36mteslamate_1  |[0m 2022-04-22 23:06:01.872 [error] POST https://auth.tesla.com/oauth2/v3/token -> error: "non-existing domain" (8014.368 ms)
[36mteslamate_1  |[0m 2022-04-22 23:06:01.873 [warning] Token refresh failed: %TeslaApi.Error{
[36mteslamate_1  |[0m   env: nil,
[36mteslamate_1  |[0m   message: "non-existing domain",
[36mteslamate_1  |[0m   reason: :token_refresh
[36mteslamate_1  |[0m }
[36mteslamate_1  |[0m 2022-04-22 23:06:01.873 [warning] Retrying in 5 minutes...
[36mteslamate_1  |[0m 2022-04-22 23:06:36.723 [error] GET https://owner-api.teslamotors.com/api/1/vehicles/1492656516736953 -> error: "non-existing domain" (8013.926 ms)
[36mteslamate_1  |[0m 2022-04-22 23:06:36.724 [warning] TeslaApi.Error / non-existing domain
[36mteslamate_1  |[0m 2022-04-22 23:06:36.725 car_id=1 [error] Error / :unknown
[36mteslamate_1  |[0m 2022-04-22 23:07:14.742 [error] GET https://owner-api.teslamotors.com/api/1/vehicles/1492656516736953 -> error: "non-existing domain" (8013.911 ms)
[36mteslamate_1  |[0m 2022-04-22 23:07:14.743 [warning] TeslaApi.Error / non-existing domain
[36mteslamate_1  |[0m 2022-04-22 23:07:14.743 car_id=1 [error] Error / :unknown
[36mteslamate_1  |[0m 2022-04-22 23:07:52.760 [error] GET https://owner-api.teslamotors.com/api/1/vehicles/1492656516736953 -> error: "non-existing domain" (8013.931 ms)
[36mteslamate_1  |[0m 2022-04-22 23:07:52.761 [warning] TeslaApi.Error / non-existing domain
[36mteslamate_1  |[0m 2022-04-22 23:07:52.762 car_id=1 [error] Error / :unknown
[36mteslamate_1  |[0m 2022-04-22 23:08:30.781 [error] GET https://owner-api.teslamotors.com/api/1/vehicles/1492656516736953 -> error: "non-existing domain" (8014.058 ms)
[36mteslamate_1  |[0m 2022-04-22 23:08:30.781 [warning] TeslaApi.Error / non-existing domain
[36mteslamate_1  |[0m 2022-04-22 23:08:30.781 car_id=1 [error] Error / :unknown
[36mteslamate_1  |[0m 2022-04-22 23:09:08.799 [error] GET https://owner-api.teslamotors.com/api/1/vehicles/1492656516736953 -> error: "non-existing domain" (8013.961 ms)
[36mteslamate_1  |[0m 2022-04-22 23:09:08.800 [warning] TeslaApi.Error / non-existing domain
[36mteslamate_1  |[0m 2022-04-22 23:09:08.800 car_id=1 [error] Error / :unknown
[36mteslamate_1  |[0m 2022-04-22 23:09:46.818 [error] GET https://owner-api.teslamotors.com/api/1/vehicles/1492656516736953 -> error: "non-existing domain" (8013.961 ms)
[36mteslamate_1  |[0m 2022-04-22 23:09:46.819 [warning] TeslaApi.Error / non-existing domain
[36mteslamate_1  |[0m 2022-04-22 23:09:46.819 car_id=1 [error] Error / :unknown
[36mteslamate_1  |[0m 2022-04-22 23:10:24.837 [error] GET https://owner-api.teslamotors.com/api/1/vehicles/1492656516736953 -> error: "non-existing domain" (8013.943 ms)
[36mteslamate_1  |[0m 2022-04-22 23:10:24.838 [warning] TeslaApi.Error / non-existing domain
[36mteslamate_1  |[0m 2022-04-22 23:10:24.838 car_id=1 [error] Error / :unknown
[36mteslamate_1  |[0m 2022-04-22 23:11:01.874 [info] Refreshing access token ...
[36mteslamate_1  |[0m 2022-04-22 23:11:02.855 [error] GET https://owner-api.teslamotors.com/api/1/vehicles/1492656516736953 -> error: "non-existing domain" (8013.736 ms)
[36mteslamate_1  |[0m 2022-04-22 23:11:02.856 [warning] TeslaApi.Error / non-existing domain
[36mteslamate_1  |[0m 2022-04-22 23:11:02.856 car_id=1 [error] Error / :unknown
[36mteslamate_1  |[0m 2022-04-22 23:11:09.891 [error] POST https://auth.tesla.com/oauth2/v3/token -> error: "non-existing domain" (8015.027 ms)
[36mteslamate_1  |[0m 2022-04-22 23:11:09.892 [warning] Token refresh failed: %TeslaApi.Error{
[36mteslamate_1  |[0m   env: nil,
[36mteslamate_1  |[0m   message: "non-existing domain",
[36mteslamate_1  |[0m   reason: :token_refresh
[36mteslamate_1  |[0m }
[36mteslamate_1  |[0m 2022-04-22 23:11:09.892 [warning] Retrying in 5 minutes...
```

Then it almost looks as if you restart the Pi or docker containers:
```
[36mteslamate_1  |[0m 2022-04-23 11:00:12.489 car_id=1 [error] Error / :unknown
[36mteslamate_1  |[0m 2022-04-23 11:00:50.507 [error] GET https://owner-api.teslamotors.com/api/1/vehicles/1492656516736953 -> error: "non-existing domain" (8013.777 ms)
[36mteslamate_1  |[0m 2022-04-23 11:00:50.508 [warning] TeslaApi.Error / non-existing domain
[36mteslamate_1  |[0m 2022-04-23 11:00:50.509 car_id=1 [error] Error / :unknown
[36mteslamate_1  |[0m waiting for postgres at database:5432
[36mteslamate_1  |[0m waiting for postgres at database:5432
[36mteslamate_1  |[0m waiting for postgres at database:5432
[36mteslamate_1  |[0m waiting for postgres at database:5432
[36mteslamate_1  |[0m 2022-04-23 10:17:56.874 [info] Migrations already up
[36mteslamate_1  |[0m 2022-04-23 10:18:06.326 [info] System Info: Erlang/OTP 24 (emu)
[36mteslamate_1  |[0m 2022-04-23 10:18:06.327 [info] Version: 1.26.1
[36mteslamate_1  |[0m 2022-04-23 10:18:14.769 [error] POST https://auth.tesla.com/oauth2/v3/token -> error: "non-existing domain" (8172.368 ms)
[36mteslamate_1  |[0m 2022-04-23 10:18:14.771 [warning] Token refresh failed: %TeslaApi.Error{
[36mteslamate_1  |[0m   env: nil,
[36mteslamate_1  |[0m   message: "non-existing domain",
[36mteslamate_1  |[0m   reason: :token_refresh
[36mteslamate_1  |[0m }
[36mteslamate_1  |[0m 2022-04-23 10:18:14.772 [info] Scheduling token refresh in 7 min
[36mteslamate_1  |[0m 2022-04-23 10:18:14.831 [info] Running TeslaMateWeb.Endpoint with cowboy 2.9.0 at :::4000 (http)
[36mteslamate_1  |[0m 2022-04-23 10:18:14.847 [info] Access TeslaMateWeb.Endpoint at http://localhost
[36mteslamate_1  |[0m 2022-04-23 10:18:22.871 [error] GET https://owner-api.teslamotors.com/api/1/vehicles -> error: "non-existing domain" (8015.045 ms)
[36mteslamate_1  |[0m 2022-04-23 10:18:22.871 [warning] TeslaApi.Error / non-existing domain
[36mteslamate_1  |[0m 2022-04-23 10:18:22.872 [warning] Could not get vehicles: :unknown
[36mteslamate_1  |[0m 2022-04-23 10:18:22.928 [warning] Using fallback vehicles:
[36mteslamate_1  |[0m 
[36mteslamate_1  |[0m [
[36mteslamate_1  |[0m   %TeslaApi.Vehicle{
[36mteslamate_1  |[0m     api_version: nil,
[36mteslamate_1  |[0m     backseat_token: nil,
[36mteslamate_1  |[0m     backseat_token_updated_at: nil,
[36mteslamate_1  |[0m     calendar_enabled: nil,
[36mteslamate_1  |[0m     charge_state: nil,
[36mteslamate_1  |[0m     climate_state: nil,
[36mteslamate_1  |[0m     color: nil,
[36mteslamate_1  |[0m     display_name: "Schtroumpf Electrique",
[36mteslamate_1  |[0m     drive_state: nil,
[36mteslamate_1  |[0m     gui_settings: nil,
[36mteslamate_1  |[0m     id: 1492656516736953,
[36mteslamate_1  |[0m     in_service: false,
[36mteslamate_1  |[0m     option_codes: [],
[36mteslamate_1  |[0m     state: "unknown",
[36mteslamate_1  |[0m     tokens: [],
[36mteslamate_1  |[0m     vehicle_config: nil,
[36mteslamate_1  |[0m     vehicle_id: 276117568696,
[36mteslamate_1  |[0m     vehicle_state: nil,
[36mteslamate_1  |[0m     vin: "5YJ3E1EA6NF152959"
[36mteslamate_1  |[0m   }
[36mteslamate_1  |[0m ]
[36mteslamate_1  |[0m 2022-04-23 10:18:22.928 [info] Starting logger for 'Schtroumpf Electrique'
[36mteslamate_1  |[0m 2022-04-23 10:18:23.133 [info] MQTT connection has been established
```

**Reply by manikoutai** [reply]
That's interesting.

First, the "Getting OFFLINE while in drive:

- This shows a difference in the time observed in the log and what really occurred.  At "14:45hrs" I drove from the hotel street INSIDE of the underground parking.  So that's why we see the car getting OFFLINE while driving. But in the log file this occurred at 18:45hrs, so I need to investigate why there is a 4hrs difference between the actual time and the time reported by teslamate. (let me know if you have something to propose).

It looks to me that teslamate is not able to refresh the token while the car was OFFLINE for 30 minutes could be a bug on teslamate.  The OFFLINE state was there for 24hrs.  Only a restart of the raspberry pi resolved the situation.

The raspberry pi is dedicated to teslamate. I have installed all the requirements to have teslamate running.  I have also installed something to access it via Windows Remote desktop.  I have a daily backup that runs at 5 am every day and that was not interrupted.

Who could have this situation approved as an official bug?

**Reply by cwanja** [reply]
>So that's why we see the car getting OFFLINE while driving. But in the log file this occurred at 18:45hrs, so I need to investigate why there is a 4hrs difference between the actual time and the time reported by teslamate. (let me know if you have something to propose).

What is the `TZ` environment variable in your YML file? 

> It looks to me that teslamate is not able to refresh the token while the car was OFFLINE for 30 minutes could be a bug on teslamate. The OFFLINE state was there for 24hrs. Only a restart of the raspberry pi resolved the situation.

I doubt this. As every other time your installation as able to refresh the token just fine. I do not think TeslaMate needs to talk to your car to refresh the token, but that is an Adrian question. He is also the only one who can deem it a bug. You can submit an issue, he can review it and move it back to discussion if he does not feel like it is a bug. I do not think it is a bug.

**Reply by manikoutai** [reply]
@cwanja Thanks a lot for all your help here.  This is the first time I post about teslamate and first time I try to look at the logs to interpret them, so your feedback was precious.  I have logged a new bug, let's see if I receive some feedback about it.  Unfortunately I won't be able to reproduce at will, only the log will be useful.

I don't see any entries about "TZ" in the YML file.  Do you know what should be the entry name and value for Montreal/New-York?

**Reply by cwanja** [reply]
Happy to help @manikoutai - sorry I can not be more prescriptive in a solution.

>I don't see any entries about "TZ" in the YML file. Do you know what should be the entry name and value for Montreal/New-York?

You will need to add it. https://docs.teslamate.org/docs/configuration/environment_variables

Looks like `CA`. https://en.wikipedia.org/wiki/List_of_tz_database_time_zones. 

Once you add it, restart the container and you can verify that the right timezone for the logs is applied and matches your current timezone.

**Comment by ktsenvolt** [comment]
Have you try to reboot Raspberry pi?

**Reply by manikoutai** [reply]
I have restarted the raspberry pi on April 23th around 7:45am, which resolve the situation after losing a full day of driving data ( ~500km).  Such an operation was only possible once at home, and if my trip was longer I would have lost several other days of data I believe.

**Comment by ktsenvolt** [comment]
Why don't you get a Dynamic DNS from "my.noip.com"(It's Free) and you can check your TeslaMate activity outside of your home? Or you can use VNC to control your Raspberry.

**Reply by manikoutai** [reply]
It is in my "to do" list actually and I need to first identify the safest approach before exposing this outside of the house.  I don't know if Chrome remote desktop could be a good approach or not, knowing that my raspberry pi is very old (2015).

I have already something that allow me to reach to my raspberry pi through Windows remote desktop, which I use from other PC.  I need to check if an Andoid client exist for it and if it would be a practical solution to access my machine from outside of the house. So basically, 3 solutions to investigate:
1) VNC
2) Chrome Remote Desktop
3) Windows remote desktop

**Reply by ktsenvolt** [reply]
Today I am having a similar issue. From 9:AM to 12:15 P.M. The car could not keep track of charging and driving. Is it possible Tesla server was down? Because at 9:00 A.M. I could not get into Tesla App. Please let me know how to get into Teslamate log file? (I don't know how to post the logs)

Thanks.

I am using Raspberry Pi 4. Advanced installation with Traefik, Let's Encrypt & HTTP Basic Auth

**Reply by cwanja** [reply]
> Please let me know how to get into Teslamate log file? (I don't know how to post the logs)

`docker-compose logs teslamate`

Find the timestamp that is associated with the issues. If your containers timezones are not the same as your machines timezone, then there will be a difference.

**Reply by ktsenvolt** [reply]
I am with New York time zone. I have tried "docker-compose logs teslamate" and how to do the screen dump? How to download it to log file? Thank you.

**Reply by cwanja** [reply]
> I am with New York time zone. I have tried "docker-compose logs teslamate" and how to do the screen dump? How to download it to log file? Thank you.

Select the text from the terminal window. Copy. Paste into a reply here.

**Reply by ktsenvolt** [reply]
Thank you and have a nice day.
